e2e/test-runner/wsl_tester.go

```diff
@@ -0,0 +1,82 @@
+package test_runner
+
+import (
+	"flag"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+)
+
+// Defaults choosen based on the default state of the repository on git-clone.
+const defaultLauncherName = "ubuntudev.launchername.dev.exe"
+const defaultDistroName = "UbuntuDev.WslId.Dev"
+
+const serverLogPath = "/var/log/installer/systemsetup-server-debug.log"
+const clientLogPath = "ubuntu-desktop-installer/packages/ubuntu_wsl_setup/build/windows/runner/Debug/.ubuntu_wsl_setup.exe/ubuntu_wsl_setup.exe.log"
+
+var launcherName = flag.String("launcher-name", defaultLauncherName, "WSL distro launcher under test.")
+var distroName = flag.String("distro-name", defaultDistroName, "WSL distro instance registered for testing.")
+
+// A testing wrapper with functions simplifying composing and running commands through os.Exec and checking their outputs.
+type Tester struct {
+	*testing.T
+}
+
+func checkTestEnvMinRequirements(t *testing.T) {
+	rootDir := os.Getenv("GITHUB_WORKSPACE")

```

Comment:
`$GITHUB_WORKSPACE ` is a well-known environment variable in the context of GitHub actions, set by the checkout action, which is usually the first step of any workflow. Calling it something else is not a problem but it requires the CI workflow to create a new environment variable with the same value as this one. I'd expect the check (and error message) below to be enough to clarify the situation when this variable is not set, since it aborts the test in the very beginning.

Even though, I'm not exactly disagreeing with you.